src/main/java/carpet/script/annotation/Locator.java

```diff
@@ -0,0 +1,270 @@
+         * located function requires. Note that FunctionLocators consume all remaining arguments even if this is set to {@code false}.</p>
+         */
+        boolean checkArgs();
+    }
+
+    /**
+     * <p>Class that holds locators and methods to get them</p>
+     * 
+     * <p>Not part of the public API, just that interfaces must have all members public</p>
+     */
+    static final class Locators
+    {
+        private Locators() {}
+
+        static <R> ValueConverter<R> fromAnnotatedType(AnnotatedType annoType, Class<R> type)
+        {
+            if (annoType.isAnnotationPresent(Block.class))
+                return new BlockLocator<R>(annoType.getAnnotation(Block.class), type);
+            if (annoType.isAnnotationPresent(Function.class))
+                return new FunctionLocator<R>(annoType.getAnnotation(Function.class), type);
+            if (annoType.isAnnotationPresent(Vec3d.class))
+                return new Vec3dLocator<R>(annoType.getAnnotation(Vec3d.class), type);
+            throw new IllegalStateException("Locator#fromAnnotatedType got called with an incompatible AnnotatedType");
+        }
+
+        private static class BlockLocator<R> extends AbstractLocator<R>
+        {
+            private final boolean returnBlockValue;
+            private final boolean acceptString;
+            private final boolean anyString;
+            private final boolean optional;
+
+            public BlockLocator(Block annotation, Class<R> type)
+            {
+                this.acceptString = annotation.acceptString();
+                this.anyString = annotation.anyString();
+                this.optional = annotation.optional();
+                this.returnBlockValue = type == BlockValue.class;
+                if (returnBlockValue && (anyString || optional))
+                    throw new IllegalArgumentException("Can only use anyString or optional parameters of Locator.Block if targeting a BlockArgument");
+                if (!returnBlockValue && type != BlockArgument.class)
+                    throw new IllegalArgumentException("Locator.Block can only be used against BlockArgument or BlockValue types!");
+            }
+
+            @Override
+            public String getTypeName()
+            {
+                return "block";
+            }
+
+            @Override
+            public R checkAndConvert(Iterator<Value> valueIterator, Context context, Context.Type theLazyT)
+            {
+                BlockArgument locator = null; // BlockArgument.findIn((CarpetContext)context, valueIterator (requires changing to Value), 0,
+                                              // acceptString, optional, anyString);
+                // return (R) (returnBlockValue ? locator.block : locator);
+                throw new NotImplementedException("Locator.Block still requires adapting BlockArgument to accept iterators (which is actually simple)");
```

Comment:
Oh, so its only a wrapper essentially. I assume you would want to implement that